etc/sync-function-template.js

```diff
@@ -36,63 +38,134 @@ function synctos(doc, oldDoc) {
     }
   }
 
-  // A document definition may define its channels for each operation (view, add, replace, delete) as either a string or an array of
-  // strings. In either case, add them to the list if they are not already present.
-  function appendToChannelList(allChannels, channelsToAdd) {
-    if (!isValueNullOrUndefined(channelsToAdd)) {
-      if (channelsToAdd instanceof Array) {
-        for (var i = 0; i < channelsToAdd.length; i++) {
-          var channel = channelsToAdd[i];
-          if (allChannels.indexOf(channel) < 0) {
-            allChannels.push(channel);
+  // Retrieves the old doc's effective value. If it is null, undefined or its "_deleted" property is true, returns null. Otherwise, returns
+  // the value of the "oldDoc" parameter.
+  function getEffectiveOldDoc(oldDoc) {
+    return (oldDoc && !(oldDoc._deleted)) ? oldDoc : null;
+  }
+
+  // A document definition may define its authorizations (channels, roles or users) for each operation type (view, add, replace, delete or
+  // write) as either a string or an array of strings. In either case, add them to the list if they are not already present.
+  function appendToAuthorizationList(allAuthorizations, authorizationsToAdd) {
```

Comment:
Performs the same function as `appendToChannelList` but now with a generic name that allows it to be used for channels, roles, etc.